hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java

```diff
@@ -144,13 +171,35 @@ public void stop() {
     if (executorService != null) {
       executorService.shutdown();
     }
-    if (hiveLocalDir != null) {
-      hiveLocalDir.delete();
-    }
     if (baseHandler != null) {
       baseHandler.shutdown();
     }
     METASTORE_THREADS_SHUTDOWN.invoke();
+    HMS_HANDLER_THREAD_LOCAL_CONF.remove();
+    HMS_HANDLER_THREAD_LOCAL_TXN.remove();
+    Object connPool = CONN_POOL.get();
```

Comment:
These should be cleaned up immediately when there is a configuration change in the connection pool URI.
Maybe our tests are not setting the connection pool URI correctly?

Reply:
`TxnHandler::setupJdbcConnectionPool` doesn't do anything if `connPool` is already set:
https://github.com/apache/hive/blob/release-1.2.1/metastore/src/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java#L1862

Besides, I also want to close it in case there's any resource leak.

Reply:
So again, we face the issue that the HiveMetastore is not planned for stop/start (which is a shame), but IMHO we could/should not fix it here. We just have to use it in the way as it is planned:
- Create only one HMS instance per test class
- Make sure that the test infra is using different thread (or JVM) to run the test classes
- Remove other leftover resources which are not cleaned up (like the temp directory you correctly identified)

Reply:
I didn't find gradle/junit configurations to force test classes to run in different threads. We can go with different JVMs, at the cost of longer test time. Actually that's what I did when I worked on the hive integration in flink. I believe it can be achieved by setting `forkEvery` to 1.

Reply:
I think(hope) it does this automatically, but not a gradle expert myself. I remember comping to this conclusion when I have faced some ThreadLocal issues myself, but I might be mistaken